ocp-templates/scripts/import_ocp_project_metadata.sh

```diff
@@ -486,19 +480,25 @@ do
 		if [[ $ocp_proj_namespace_suffix == "cd" ]]; then 
 			echo "--  not creating pull secret to OCP, this is CD - just adding image pull rights"
 			oc policy add-role-to-user system:image-puller system:serviceaccount:${project_name}-cd:jenkins -n cd
-		else
-			if [[ ! -z ${OD_OCP_SOURCE_TOKEN} ]]; then 
-				echo "Creating OCP OD pull secret"
-				oc create secret docker-registry odocp  --docker-server=${OD_OCP_DOCKER_REGISTRY_SOURCE_HOST} --docker-username=cd/cd-integration --docker-password=${OD_OCP_SOURCE_TOKEN} --docker-email=a@b.com     
-				oc secrets link deployer odocp --for=pull                                   
-				oc secrets link default odocp --for=pull
-			else
-				echo "OCP OD Token not set - assuming local build"
-			fi
 		fi
 	else
 		echo "!!! Project ${curr_ocp_namespace} already exists - skipping creation"
     fi
+
+	if [[ ! -z ${OD_OCP_SOURCE_TOKEN} ]] && [[ ! "$ocp_proj_namespace_suffix" == "cd" ]]; then 
+		echo "Creating OCP OD pull secret for ${OD_OCP_DOCKER_REGISTRY_SOURCE_HOST}"
+		oc create secret docker-registry odocp  --docker-server=${OD_OCP_DOCKER_REGISTRY_SOURCE_HOST} --docker-username=cd/cd-integration --docker-password=${OD_OCP_SOURCE_TOKEN} --docker-email=a@b.com     
+		oc secrets link deployer odocp --for=pull                                   
+		oc secrets link default odocp --for=pull
+	else
+		echo "OCP OD Token not set - assuming local build"
+	fi
+
+	# add admins
+	for admin_user in $(echo $OD_PRJ_ADMINS | sed -e 's/,/ /g');
+	do		
+		oc policy add-role-to-user admin ${admin_user}
+	done
```

Comment:
So if I understand correctly, this will now run also for existing projects. Will this break for existing projects that have those things already? Or am I on the wrong track here? ...

Reply:
<pre>
utschig@WKS0000627058 MINGW64 /c/code_bix/opendevstack_at_BIX/ods-project-quickstarters (bug/import_docker_secret)
$ oc policy add-role-to-user admin utschig
role "admin" added: "utschig"

utschig@WKS0000627058 MINGW64 /c/code_bix/opendevstack_at_BIX/ods-project-quickstarters (bug/import_docker_secret)
$ oc policy add-role-to-user admin utschig
role "admin" added: "utschig"
</pre>

the call is idempotent .. :)

Reply:
I meant the whole block - is e.g. `oc create secret docker-registry` idempotent?